Cython 0.29

Cyanoxygen opened the pull request:
Topic Description
-----------------

This PR introduces the legacy version of Cython (0.29.x branch, needed for older sources) and also updates Cython 3.0.x to its latest patch version.

Package(s) Affected
-------------------

### New

- `cython-0.29` 0.29.36

### Existing

- `cython` 3.0.2
- `rapidfuzz` 3.2.0-1


Security Update?
----------------

No

Build Order
-----------

```
cython cython-0.29 rapidfuzz
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

Cyanoxygen committed:
cython-0.29: new, 0.29.36

- This package is intended to provide compatibility to programs written
  for Cython 0.29. It can not coexist with Cython 3.0 onwards.

Cyanoxygen committed:
cython: update to 3.0.2

Cyanoxygen committed:
cython: mark conflict between cython-0.29 and cython

- As they are incompatible and can not coexist with each other.

Cyanoxygen committed:
rapidfuzz: (Arch Linux) remove version constraints

- Cython is pinned to 3.0.0, while the latest Cython release with 3.0 is
  3.0.2.

MingcongBai commented on lang-python/cython-0.29/autobuild/build:
```diff
@@ -0,0 +1,9 @@
+python3 setup.py install --root="$PKGDIR"
```
Missing abinfo.

MingcongBai commented on lang-python/cython-0.29/autobuild/defines:
```diff
@@ -0,0 +1,9 @@
+PKGDEP="python-2 python-3 setuptools"
+
+PKGPROV=cython
+PKGCONFL="cython>=3.0.0"
```
No, it's in fact conflicting with all `cython` versions, since older `cython` versions has the same file lists as `cython-0.29`.

MingcongBai committed:
cython-0.29: conflict with all cython versions

- There is no point in setting a version boundary there are file conflicts.
- Also, no point in renaming the Cython executables since, again, files will
  conflict between the two packages no matter what.
- Drop PKGPROV as this package does not provide an equivalent for cython.

MingcongBai committed:
cython: fix typo PKGBCONFL => PKGCONFL

MingcongBai committed:
cython: drop unneeded executable renaming

There will be file conflicts with cython-0.29 no matter what.
